### PR TITLE
Add mute and unmute formats for Volume and PulseVolume widgets

### PR DESCRIPTION
**Pull request by shyguyCreate**
Rewrite of #4151

### CODE REVIEW

**Comment by shyguyCreate** [comment]
Help. Can someone point me out how to get the PulseAudio volume value when mute, it always returns `-1`.

I can't find anything in [pulse_volume.py](https://github.com/qtile/qtile/blob/master/libqtile/widget/pulse_volume.py).

File: libqtile/widget/volume.py [hunk 1]
```diff
@@ -221,7 +224,9 @@ def get_volume(self):
             check_mute = subprocess.getoutput(self.check_mute_command)
 
         if self.check_mute_string in check_mute:
-            return -1
+            self.mute = 1
+        else:
+            self.mute = 0
```
Comment (tych0) [inline libqtile/widget/volume.py:229]: Can you use python's True + False for these literals?

Comment (tych0) [reply libqtile/widget/volume.py:229]: Better yet, can you make `get_volume()` return a tuple that's the volume and whether or not the widget is muted? You can override it in your test with some monkey patching.

**Comment by elParaguayo** [comment]
Agree with @tych0 - this needs some work. Having `get_volume()` return a tuple of the volume level and whether the device is muted or not seems like the correct approach.

**Review by elParaguayo — CHANGES_REQUESTED** [review]
I think this would be a nice addition to the code. However, I'm marking this one as "request changes` for now because:
a) it needs changes as suggested by tych0
b) the volume widgets are likely to be used by a large portion of the userbase so I want to take a close look before this gets merged.

**Comment by elParaguayo** [comment]
Why close this?

**Comment by shyguyCreate** [comment]
Sorry I panic, I think I messed up the commit history.

**Comment by shyguyCreate** [comment]
For the record, I did messed up the history right? Because I see 10+ commits that are not mine.

**Comment by elParaguayo** [comment]
If you look at this PR page you'll see it says it's only adding one commit. The other ones were probably because you did a merge to bring the branch in line with the current master. However, github knows those commits aren't needed here.

**Comment by shyguyCreate** [comment]
Next time I have to update the master I'll ask before undoing everything.

**Comment by elParaguayo** [comment]
Just rebase onto master rather than try to merge changes into the branch.

**Comment by shyguyCreate** [comment]
Question: I have heard bad things about using `--force` on push, and after playing with git rebase and getting my commit on top, I really need to use `--force` with git push?

**Comment by elParaguayo** [comment]
There are definitely times when you shouldn't use force (eg when pushing to a branch that other users are tracking and making changes to). However, for a PR here, we're happy for you to use force.

**Comment by shyguyCreate** [comment]
All right, thanks

File: libqtile/widget/volume.py [hunk 2]
```diff
@@ -214,21 +217,20 @@ def get_volume(self):
 
             mixer_out = subprocess.getoutput(get_volume_cmd)
         except subprocess.CalledProcessError:
-            return -1
+            return -1, 0
```
Comment (tych0) [inline libqtile/widget/volume.py:220]: Can you switch to True/False here? Other than that looks good to me.

Comment (shyguyCreate) [reply libqtile/widget/volume.py:220]: What do you mean exactly? Do you want to use True/False in self.mute instead of 1/0?

Comment (tych0) [reply libqtile/widget/volume.py:220]: yep, exactly

Comment (shyguyCreate) [reply libqtile/widget/volume.py]: Done

**Comment by tych0** [comment]
Thanks!

**Comment by elParaguayo** [comment]
This broke the Volume widget. We had a method called `mute` but this is overwritten by defining a variable called `mute` during `__init__`. 

We need better tests!

**Comment by shyguyCreate** [comment]
It should now be fixed in the new PR.

**Pull request merged**